torch/distributed/distributed_c10d.py

```diff
@@ -73,7 +76,25 @@ def __new__(cls, name):
                              "Gloo or MPI backend for collective operations "
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
-            raise ValueError("Invalid backend: '{}'".format(name))
+            try:
+                if name not in sys.modules:
+                    # using "torch_" prefix to avoid module naming conflict/confusion 
+                    other_backend = importlib.import_module("torch_{}".format(name.lower()))
+                else:
+                    other_backend = sys.modules["torch_{}".format(name.lower())]
+
+                if other_backend.__dict__.get("name"):
+                    pg_name = other_backend.name()
+                elif other_backend.__dict__.get("ProcessGroup{}".format(name.lower().capitalize())):
+                    pg_name = "ProcessGroup{}".format(name.lower().capitalize())
```

Comment:
You can generate this once, store it in a variable, and use it twice. Copying the name generation is error prone.

Reply:
done